amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/AmlEmittersHelper.scala

```diff
@@ -1,26 +1,69 @@
 package amf.plugins.document.vocabularies.emitters.instances
+import amf.client.environment.AMLConfiguration
+import amf.client.remod.parsing.AMLDialectInstanceParsingPlugin
 import amf.core.annotations.Aliases.{Alias, FullUrl, ImportLocation, RefId}
 import amf.core.annotations.{Aliases, LexicalInformation}
 import amf.core.emitter.BaseEmitters.traverse
 import amf.core.emitter.{EntryEmitter, SpecOrdering}
 import amf.core.model.document.{BaseUnit, DeclaresModel}
 import amf.core.model.domain.AmfObject
-import amf.core.parser.Position
+import amf.core.parser.{ParserContext, Position}
 import amf.core.parser.Position.ZERO
 import amf.plugins.document.vocabularies.AMLPlugin
 import amf.plugins.document.vocabularies.emitters.common.{ExternalEmitter, IdCounter}
 import amf.plugins.document.vocabularies.model.document.{Dialect, DialectLibrary, ExternalContext}
 import amf.plugins.document.vocabularies.model.domain.{NodeMappable, NodeMapping, UnionNodeMapping}
 import org.yaml.model.YDocument.EntryBuilder
 import amf.core.utils.Regexes.Path
+import amf.plugins.document.vocabularies.model.domain.NodeMappable.AnyNodeMappable
+import org.mulesoft.common.core.CachedFunction
+import org.mulesoft.common.functional.MonadInstances.identityMonad
 
 import scala.collection.mutable
 
-trait AmlEmittersHelper {
-  val dialect: Dialect
+trait NodeMappableFinder {
+  def findNode(id: String): Option[(Dialect, AnyNodeMappable)]
+}
+
+object DefaultNodeMappableFinder {
+  def apply(dialects: Seq[Dialect]) = new DefaultNodeMappableFinder(dialects)
+  def apply(ctx: ParserContext) = {
+    val knownDialects = ctx.config.sortedParsePlugins.collect {
+      case plugin: AMLDialectInstanceParsingPlugin => plugin.dialect
+    }
+    new DefaultNodeMappableFinder(knownDialects)
+  }
+  def apply(ctx: AMLConfiguration) = {
+    val knownDialects = ctx.registry.plugins.parsePlugins.collect {
```

Comment:
If this is only used for emission it might be more correct to search among the render plugins